lib/math/math.go

```diff
@@ -0,0 +1,150 @@
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package math provides basic constants and mathematical functions.
+package math // import "go.starlark.net/lib/math"
+
+import (
+	"fmt"
+	"math"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+// Module math is a Starlark module of math-related functions and constants.
+// The module defines the following functions:
+//
+//     abs(x) - Returns the absolute value of x.
+//     ceil(x) - Returns the ceiling of x, the smallest integer greater than or equal to x.
+//     floor(x) - Returns the floor of x, the largest integer less than or equal to x.
+//     round(x) - Returns the nearest integer, rounding half away from zero.
+//
+//     exp(x) - Returns e raised to the power x, where e = 2.718281… is the base of natural logarithms.
+//     sqrt(x) - Returns the square root of x.
+//
+//     acos(x) - Returns the arc cosine of x, in radians.
+//     asin(x) - Returns the arc sine of x, in radians.
+//     atan(x) - Returns the arc tangent of x, in radians.
+//     atan2(y, x) - Returns atan(y / x), in radians.
+//                   The result is between -pi and pi.
+//                   The vector in the plane from the origin to point (x, y) makes this angle with the positive X axis.
+//                   The point of atan2() is that the signs of both inputs are known to it, so it can compute the correct
+//                   quadrant for the angle.
+//                   For example, atan(1) and atan2(1, 1) are both pi/4, but atan2(-1, -1) is -3*pi/4.
+//     cos(x) - Returns the cosine of x, in radians.
+//     hypot(x, y) - Returns the Euclidean norm, sqrt(x*x + y*y). This is the length of the vector from the origin to point (x, y).
+//     sin(x) - Returns the sine of x, in radians.
+//     tan(x) - Returns the tangent of x, in radians.
+//
+//     degrees(x) - Converts angle x from radians to degrees.
+//     radians(x) - Converts angle x from degrees to radians.
+//
+//     acosh(x) - Returns the inverse hyperbolic cosine of x.
+//     asinh(x) - Returns the inverse hyperbolic sine of x.
+//     atanh(x) - Returns the inverse hyperbolic tangent of x.
+//     cosh(x) - Returns the hyperbolic cosine of x.
+//     sinh(x) - Returns the hyperbolic sine of x.
+//     tanh(x) - Returns the hyperbolic tangent of x.
+//
+//     log(x) - Returns the natural logarithm of x.
+//     log2(x) - Returns the decimal logarithm of x. The special cases are the same as for Log.
+//     log10(x) - Returns the natural logarithm of 1 plus its argument x. It is more accurate than Log(1 + x) when x is near.
+//
+// All function accept both int and float values as arguments.
+//
+// The module also defines approximations of the following constants:
+//
+//     e - The base of natural logarithms, approximately 2.71828.
+//     phi - The golden ratio, (1 + sqrt(5)) / 2, approximately 1.61803.
```

Comment:
Let's delete phi. It's neither well known nor widely needed, it's not part of Python's math library, and it's trivial for users to define themselves.